File: frontend/src/models/MockInterview.ts
export interface Question {
	question: string;
	answer: string;
}

export interface Skill {
	skill: string;
	level: string; // e.g., Beginner, Intermediate, Expert
	rating: number; // 0-10
}

export interface MockInterview {
	id: number;
	batch_id: number;
	candidate_id: number;
	interviewer_name?: string;
	interview_date: string;
	questions?: Question[];
	skills?: Skill[];
	feedback?: string;
	overall_rating?: number;
	status: 'pending' | 'cleared' | 're-test' | 'rejected' | 'absent';
	interview_type?: 'internal' | 'external';
	interview_category?: 'hr' | 'domain';
	start_time?: string;
	end_time?: string;
	duration_minutes?: number;
	candidate_token?: string;
	candidate_submitted_at?: string;
	interviewer_id?: number;
	created_at: string;
	updated_at: string;
	candidate?: {
		id: number;
		public_id: string;
		name: string;
		documents?: import('./candidate').CandidateDocument[];
	};
	batch?: {
		id: number;
		public_id: string;
		batch_name: string;
	};
}

export interface MockInterviewCreate {
	batch_id: number;
	candidate_id: number;
	interviewer_name?: string;
	interview_date: string;
	interview_type?: 'internal' | 'external';
	interview_category?: 'hr' | 'domain';
	questions?: Question[];
	skills?: Skill[];
	feedback?: string;
	overall_rating?: number;
	status?: string;
	start_time?: string;
	end_time?: string;
	duration_minutes?: number;
	candidate_token?: string;
	candidate_submitted_at?: string;
	interviewer_id?: number;
}

export interface MockInterviewUpdate {
	interviewer_name?: string;
	interview_date?: string;
	interview_type?: 'internal' | 'external';
	interview_category?: 'hr' | 'domain';
	questions?: Question[];
	skills?: Skill[];
	feedback?: string;
	overall_rating?: number;
	status?: string;
	candidate_token?: string;
	interviewer_id?: number;
}

File: backend/app/schemas/mock_interview.py
from datetime import datetime
from typing import Optional, List, Dict, Any, Union
from pydantic import BaseModel, root_validator

# Shared properties
class MockInterviewBase(BaseModel):
    interviewer_name: Optional[str] = None
    interview_date: Optional[datetime] = None
    questions: Optional[List[Dict[str, Any]]] = None # List of {question: str, answer: str}
    skills: Optional[List[Dict[str, Any]]] = None # List of {skill_name: str, level: str, rating: int}
    others: Optional[Dict[str, Any]] = None # Dynamic fields
    feedback: Optional[str] = None
    overall_rating: Optional[int] = None
    status: Optional[str] = "pending"

# Properties to receive on creation
class MockInterviewCreate(MockInterviewBase):
    batch_id: int
    candidate_id: int
    interview_date: datetime

# Properties to receive on update
class MockInterviewUpdate(MockInterviewBase):
    pass

# Properties shared by models stored in DB
class MockInterviewInDBBase(MockInterviewBase):
    id: int
    batch_id: int
    candidate_id: int

    class Config:
        orm_mode = True

# Properties to return to client
class MockInterview(MockInterviewInDBBase):
    pass

class MockInterviewInList(MockInterviewInDBBase):
    pass
